src/ucp/core/ucp_mm.c

```diff
@@ -150,9 +151,8 @@ static ucs_status_t ucp_mem_alloc(ucp_context_h context, size_t length,
             }
         }
 
-        /* TODO: pass addr */
-        status = uct_mem_alloc(NULL, length, uct_flags, &method, 1, mds, num_mds,
-                               name, &mem);
+        status = uct_mem_alloc(memh->address, length, uct_flags, &method, 1, mds,
```

Comment:
@shamisp I'm not sure I see you point
man mmap for MAP_FIXED flag says me a bit different things: it's less portable, overlapped parts will be discarded, etc. but it does. If the addr can't be used, mmap will fail, so ucp_mem_map should return NO_MEMORY.